### ISSUE TITLE
Some Thai language synonyms caused MDB_BAD_VALSIZE error on search

### ISSUE THREAD

**Issue by sukinoverse** [meilisearch/meilisearch#4031]
### **To Reproduce**
1. Add following synonyms to meilisearch
```json
{
    "japanese": [
        "ญี่ปุ่น",
        "ยี่ปุ่น"
    ]
}
```
2. When get index synonyms settings from meilisearch it will return
```json
{
    "japanese": [
        "ญปน",
        "ย ป  น"
    ]
}
```
This is pretty weird because `ญี่ปุ่น` and `ยี่ปุ่น` are just different on the first character, but meilisearch normalized them differently.

Never mind, let's go for the next step.

3. When perform a search with following query **will cause an error on meilisearch**
```json
{
	"q": "japanese"
}
```

```json
{
    "message": "internal: MDB_BAD_VALSIZE: Unsupported size of key/DB name/data, or wrong DUPFIXED size.",
    "code": "internal",
    "type": "internal",
    "link": "https://docs.meilisearch.com/errors#internal"
}
```

4. If I want to avoid this error, I need to remove `ยี่ปุ่น` from synonyms. I don't know why it caused this panic, but it's weird.

Noted: I cannot provide a documents data here due to personal data, if any dev need them you can email my at `sukino@miraiverse.me`. I will send it personally.

### **Expected behavior**
I can perform a search normally without any errors

#### Search error
<img width="1057" alt="Screenshot 2023-09-04 at 17 30 45" src="https://github.com/meilisearch/meilisearch/assets/9070844/56c428a2-1624-4891-9af5-d8556f4e2b83">

#### Updating index synonyms
<img width="1006" alt="Screenshot 2023-09-04 at 17 35 44" src="https://github.com/meilisearch/meilisearch/assets/9070844/ebae1b27-2fba-4458-8864-8e1423a2c0d7">

#### Check updated index synonyms
<img width="1012" alt="Screenshot 2023-09-04 at 17 35 58" src="https://github.com/meilisearch/meilisearch/assets/9070844/b8700970-006d-44a8-b567-aee336ac9b2f">

### **Meilisearch version:**
1.3.2

**Comment by Kerollmops** [comment]
Hey @sukinoverse 👋

Thank you very much for your detailed bug report!

Unfortunately, I cannot reproduce the bug using either no dataset or [the demo movies dataset](https://www.meilisearch.com/docs/learn/getting_started/quick_start#add-documents). Could you send the dataset by email?

@ManyTheFish I am pretty sure we are trying to fetch an empty key in the database, maybe during some normalization process. What do you think?

**Comment by curquiza** [comment]
Closed by #4033
